vulkaninfo/vulkaninfo.c

```diff
@@ -4542,24 +4517,19 @@ static void AppGpuDumpQueueProps(const struct AppGpu *gpu, uint32_t id, FILE *ou
                 "class='val'>%d</div>, <div class='val'>%d</div>)</summary></details>\n",
                 props.minImageTransferGranularity.width, props.minImageTransferGranularity.height,
                 props.minImageTransferGranularity.depth);
-        fprintf(out, "\t\t\t\t\t\t<details><summary>present support</summary>\n");
-        for (struct SurfaceNameWHandle *surface_stuff = gpu->inst->surface_chain; surface_stuff != NULL;
-             surface_stuff = surface_stuff->pNextSurface) {
-            fprintf(out, "\t\t\t\t\t\t\t<details><summary>%s = <div class='val'>%s</div></summary></details>\n",
-                    surface_stuff->name, surface_stuff->present_support ? "true" : "false");
+        for (int i = 0; i < gpu->inst->surface_ext_infos_count; ++i) {
+            fprintf(out, "\t\t\t\t\t\t<details><summary>present support(%s) = <div class='val'>%s</div></summary></details>\n",
```

Comment:
Would prefer if this was made into a section that listed present support for each surface type under it.
So instead of:
```
VkQueueFamilyProperties[0]:
===========================
        queueFlags         = GRAPHICS | COMPUTE | TRANSFER | SPARSE
        queueCount         = 16
        timestampValidBits = 64
        minImageTransferGranularity = (1, 1, 1)
        present support(VK_KHR_xcb_surface) = true
        present support(VK_KHR_xlib_surface) = true
```
It would be:
```
VkQueueFamilyProperties[0]:
===========================
        queueFlags         = GRAPHICS | COMPUTE | TRANSFER | SPARSE
        queueCount         = 16
        timestampValidBits = 64
        minImageTransferGranularity = (1, 1, 1)
        present support:
                VK_KHR_xcb_surface = true
                VK_KHR_xlib_surface = false
```

Reply:
I suppose.
One other thing: I wasn't sure how to inform this is the platform-agnostic query. Should something be done there?

Reply:
Though I think perhaps the html and human readable output could show only one value assuming it is same for all the WSI platforms....